datastores/gossip_kv/server/server.rs

```diff
@@ -11,40 +11,90 @@ use hydroflow::lattices::map_union::{KeyedBimorphism, MapUnionHashMap};
 use hydroflow::lattices::set_union::SetUnionHashSet;
 use hydroflow::lattices::PairBimorphism;
 use hydroflow::scheduled::graph::Hydroflow;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use tracing::{info, trace};
 
 use crate::model::{delete_row, upsert_row, Clock, Namespaces, RowKey, TableName};
 use crate::util::ClientRequestWithAddress;
 
+/// A trait that represents an abstract network address. In production, this will typically be
+/// SocketAddr.
+pub trait Address: Hash + Debug + Clone + Eq + Serialize {}
+impl<A> Address for A where A: Hash + Debug + Clone + Eq + Serialize {}
+
+/// A trait that represents a stream of client requests, originating at a specific address.
+pub trait ClientRequestStream<A>: Stream<Item = (ClientRequest, A)> + Unpin + 'static
+where
+    A: Address,
+{
+}
+
+impl<S, A> ClientRequestStream<A> for S
+where
+    S: Stream<Item = (ClientRequest, A)> + Unpin + 'static,
+    A: Address,
+{
+}
+
+/// A trait that represents a sink for sending client responses to a specific address.
+pub trait ClientResponseSink<A, E>: Sink<(ClientResponse, A), Error = E> + Unpin + 'static
+where
+    A: Address,
+{
+}
+impl<S, A, E> ClientResponseSink<A, E> for S
+where
+    S: Sink<(ClientResponse, A), Error = E> + Unpin + 'static,
+    A: Address,
+{
+}
+
+#[derive(Debug, Eq, PartialEq, Hash, Clone, Serialize, Deserialize)]
+pub struct SeedNode<A> {
+    pub id: MemberId,
+    pub address: A,
+}
+
 /// Creates a L0 key-value store server using Hydroflow.
 ///
 /// # Arguments
 /// -- `ib`: The input stream of client requests for the client protocol.
 /// -- `ob`: The output sink of client responses for the client protocol.
 /// -- `member_info`: The membership information of the server.
+/// -- `seed_nodes`: TODO: Good description here.
 ///
 /// # Generic Arguments
 /// -- `I`: The input stream type for the client protocol.
 /// -- `O`: The output sink type for the client protocol.
 /// -- `A`: The transport (address type) for the client protocol.
-pub fn server<I, O, A, E>(ib: I, ob: O, member_info: MemberData<A>) -> Hydroflow<'static>
+/// -- `E`: The error type for the output sink.
+pub fn server<I, O, A, E>(
+    ib: I,
+    ob: O,
+    member_info: MemberData<A>,
+    seed_nodes: Vec<SeedNode<A>>,
+) -> Hydroflow<'static>
 where
-    I: Stream<Item = (ClientRequest, A)> + Unpin + 'static,
-    O: Sink<(ClientResponse, A), Error = E> + Unpin + 'static,
-    A: Hash + Debug + Clone + Eq + Serialize + 'static, // "Address"
+    I: ClientRequestStream<A>,
+    O: ClientResponseSink<A, E>,
+    A: Address + 'static,
     E: Debug + 'static,
```

Comment:
optional: Could give the generic type params full names too

Reply:
Done. I removed ClientRequestStream and ClientResponseSink type aliases since they're not used elsewhere. The address type alias is used SeedNode, so the address type alias is retained.